docker/tools-rippled/Dockerfile

```diff
@@ -52,21 +46,40 @@ ENV VIRTUAL_ENV=/opt/venv
 RUN python3 -m venv ${VIRTUAL_ENV}
 ENV PATH=${VIRTUAL_ENV}/bin:${PATH}
 
-# ====================== CLANG-FORMAT IMAGE ======================
+# ====================== PRE-COMMIT IMAGE ======================
 # Note, we do not install a compiler here.
 
-FROM base AS clang-format
+FROM base AS pre-commit
 
 # This is not inherited from base image.
 ARG DEBIAN_FRONTEND=noninteractive
 
-# Install clang-format.
+# Install clang-format and pre-commit.
 ARG CLANG_FORMAT_VERSION
 ARG PRE_COMMIT_VERSION
 RUN pip install --no-cache \
     clang-format==${CLANG_FORMAT_VERSION} \
     pre-commit==${PRE_COMMIT_VERSION}
 
+# Install Node.JS, NPM, and prettier.
+ARG NODE_VERSION
+ARG NPM_VERSION
+ARG PRETTIER_VERSION
+ENV NPM_CONFIG_REGISTRY=https://registry.npmjs.org
+RUN <<EOF
+curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | \
+  gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_VERSION%%.*}.x nodistro main" | \
+ tee /etc/apt/sources.list.d/nodesource.list
+apt-get update
+apt-get install -y --no-install-recommends nodejs=${NODE_VERSION}-1nodesource1
```

Comment:
This is so much better than e.g. `curl https:// . . . . | bash` which I think was used to install NVM. Nice, and I am glad it works !